src/IO/H5/Python/CombineH5DatSegments.py

```diff
@@ -0,0 +1,95 @@
+# Distributed under the MIT License.
+# See LICENSE.txt for details.
+
+import logging
+import os
+import shutil
+
+import click
+import h5py
+
+
+def combine_h5_dat_segments(output, input_files):
+    """Combines multiple HDF5 dat files
+
+    This executable is used for combining a series of HDF5 files, each
+    containing one or more dat files, into a single HDF5 file. A typical
+    use case is to join dat-containing HDF5 files from different segments
+    of a simulation, with each segment containing values of the dat files
+    during different time intervals."""
+    if len(input_files) == 0:
+        logging.warning("CombineH5DatSegments: no input files, nothing to do")
+        return
+
+    # Copy first input file to output file
+    output_path = output + ".h5"
```

Comment:
```suggestion
    if not output.endswith(".h5"):
        output += ".h5"
```

then delete the `(omit file extension)` part of the help string